guide/creating-your-bot/parsing-mention-arguments.md

```diff
@@ -0,0 +1,219 @@
+
+	// Remove the prefix from the message.
+	const withoutPrefix = message.content.slice(config.prefix.length);
+	// Split the message by spaces
+	const split = withoutPrefix.split(/ +/);
+	// The first element is the command, the rest are the arguments.
+	const command = split[0];
+	const args = split.slice(1);
+
+	// ....Some commands here....
+});
+```
+
+Now you can easily test the waters by upgrading the avatar command from last time.
+This is what we have so far. It is pretty simple, it will show the avatar of who used the command.
+
+```js
+if (command === 'avatar') {
+	// What to put here to use the mention?
+	const user = message.author;
+
+	return message.channel.send(`${user.username}'s avatar: ${user.displayAvatarURL}`);
+}
+```
+
+But how do you actually get the correct user now? Well, this requires a few simple steps.  
+Putting it into a function will make it easily reusable. We will use the name `getUserFromMention` here.
+
+```js
+function getUserFromMention(mention) {
+	// Check if `mention` contains a string.
+	// If not, just return undefined to signal there is no user to be found.
+	if (!mention) return undefined;
+
+	// A user mention starts with <@ as well as a > at the end,
+	// so check for those.
+	if (mention.startsWith('<@') && mention.endsWith('>')) {
+		// Now remove the <@ and >, they are not necessary.
+		mention = mention.slice(2, -1);
+
+		// If the user has a nickname they will also have a ! in their mention.
+		// Remove that as well.
+		if (mention.startsWith('!')) {
+			mention = mention.slice(1);
+		}
+
+		// Now all that is left should be the ID,
+		// which can be used to get the user object.
+		const user = client.users.get(mention);
+		// The ID might be invalid, so check for that as well.
+		if (!user) {
+			// Just return undefined to indicate nothing was found.
+			return undefined;
+		}
+		else {
+			// A user was found, so return that.
+			return user;
+		}
+	}
+
+	// Return undefined here as well.
+	// If the code reaches this point nothing was found.
+	return undefined;
+}
+```
+
+<tip>The `slice` method is used in a more advance way here. You can read the [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/slice) for more info.</tip>
+
+Now you have a nifty function you can use whenever you need to convert a raw mention into a proper user object.
+Plugging it into the command will give you this:
+
+```js
+if (command === 'avatar') {
+	// First check if the user even tried to supply a mention arg
+	if (args[0]) {
+		// If he did try to turn that mention into a user object.
+		// The first argument should be the mention, so pass it to the function.
+		const user = getUserFromMention(args[0]);
+
+		// The function can return undefined, in which case notify
+		// the user that the argument was invalid.
+		if (!user) {
+			return message.reply('Please use a proper mention if you want to see someone else\'s avatar.');
+		}
+		else {
```

Comment:
No need for `else` here or below this one, since there's a `return` in the if statement and nothing else below the `else` blocks.